### PR TITLE
Public Links (rework) + reva update

### PR DESCRIPTION
**Pull request by refs**
this is a rework of https://github.com/owncloud/ocis-reva/pull/133. My local environment was too outdated. Instead I applied a diff on a working state and updated my branch with master.

### CODE REVIEW

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/flagset/storagepubliclink.go  1
- pkg/command/storagepubliclink.go  8
         

Clones added
============
- pkg/command/gateway.go  1
- pkg/flagset/storagepubliclink.go  4
- pkg/command/frontend.go  1
- pkg/command/storagepubliclink.go  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=5602851&bid=18353755)

**Comment by PVince81** [comment]
I bisected reva commits and the CI failure is actually happening after the public share commit:
```
@api @TestAlsoOnExternalUserBackend @files_sharing-app-required
Feature: sharing

  Background:                                                                                       # /srv/www/htdocs/owncloud/tests/acceptance/features/apiShareManagementBasic/createShare.feature:4                                                                        
    Given user "Alice" has been created with default attributes and without skeleton files          # FeatureContext::userHasBeenCreatedWithDefaultAttributesAndWithoutSkeletonFiles()                                                                                        
    And user "Alice" has uploaded file with content "ownCloud test text file 0" to "/textfile0.txt" # FeatureContext::userHasUploadedAFileWithContentTo()                                                                                                                     

  Scenario Outline: Creating a share of a file with a user and asking for various permission combinations                          # /srv/www/htdocs/owncloud/tests/acceptance/features/apiShareManagementBasic/createShare.feature:35                                        
    Given using OCS API version "<ocs_api_version>"                                                                                # FeatureContext::usingOcsApiVersion()                                                                                                     
    And user "Brian" has been created with default attributes and without skeleton files                                           # FeatureContext::userHasBeenCreatedWithDefaultAttributesAndWithoutSkeletonFiles()                                                         
    When user "Alice" shares file "textfile0.txt" with user "Brian" with permissions <requested_permissions> using the sharing API # FeatureContext::userSharesFileWithUserUsingTheSharingApi()                                                                               
    Then the OCS status code should be "<ocs_status_code>"                                                                         # OCSContext::theOCSStatusCodeShouldBe()                                                                                                   
    And the HTTP status code should be "200"                                                                                       # FeatureContext::thenTheHTTPStatusCodeShouldBe()                                                                                          
    And the fields of the last response should include                                                                             # FeatureContext::checkFields()
      | share_with             | Brian                 |
      | share_with_displayname | Brian Murphy          |
      | file_target            | /textfile0.txt        |
      | path                   | /textfile0.txt        |
      | permissions            | <granted_permissions> |
      | uid_owner              | Alice                 |
      | displayname_owner      | Alice Hansen          |
      | item_type              | file                  |
      | mimetype               | text/plain            |
      | storage_id             | ANY_VALUE             |
      | share_type             | user                  |

    Examples:
      | ocs_api_version | requested_permissions | granted_permissions | ocs_status_code |
      | 1               | 31                    | 19                  | 100             |
        │ file_target has unexpected value ''
        │ 
        file_target doesn't have value '/textfile0.txt'
        Failed asserting that false is true.

--- Failed scenarios:

    /srv/www/htdocs/owncloud/tests/acceptance/features/apiShareManagementBasic/createShare.feature:56
```

**Comment by PVince81** [comment]
I was not able to pinpoint precise commits:
```
There are only 'skip'ped commits left to test.
The first bad commit could be any of:
9b6cc77b8a1c3cdc9cf536bd9c0cf97e560db0ab
b7499d57054eefebd250e4b9415f05e1d548f4b3
613e636cb36fe0cba03ffeeca6f2212dd1538d0c
a7bb4e6c3fae7e146f33f189557742ae64c37ac3
5f8a0df10c0fccdcd06f9cbaeb69de02836ab90d
9cdada675bb0ef2a67cd270a27d66ee5688e348b
bee35adced3a0c27f6aaa1fdcf34215f3d4e8de4
c1cb7935d892d27ec8a94b58627971a105424496
41676b31f26e155ca3d1b9fd68987f6827750d3f
bb642fa185116869a3bd0e6bbc852eb6491fad6c
2ad8bf114fd75e110d49e81f293a62aa9864963d
7ac248a5e4c8fa0e242d9be3b51651dd6db4d40c
8503ee313d3f5d75eb6253c15dea840bbe379b0a
c3e9ef41602060810b6e3dc6c8383be860109281
6e5318722abf04ab7de14501ebab1308497f2a45
We cannot bisect more!
```

I had to skip a lot due to compilation errors like:
```
# github.com/owncloud/ocis-reva/pkg/command
../ocis-reva/pkg/command/authbasic.go:120:6: undefined: "github.com/cs3org/reva/cmd/revad/runtime".RunWithOptions
../ocis-reva/pkg/command/authbasic.go:123:7: undefined: "github.com/cs3org/reva/cmd/revad/runtime".WithLogger
../ocis-reva/pkg/command/authbearer.go:105:6: undefined: "github.com/cs3org/reva/cmd/revad/runtime".RunWithOptions
../ocis-reva/pkg/command/authbearer.go:108:7: undefined: "github.com/cs3org/reva/cmd/revad/runtime".WithLogger
../ocis-reva/pkg/command/frontend.go:206:6: undefined: "github.com/cs3org/reva/cmd/revad/runtime".RunWithOptions
../ocis-reva/pkg/command/frontend.go:209:7: undefined: "github.com/cs3org/reva/cmd/revad/runtime".WithLogger
../ocis-reva/pkg/command/gateway.go:165:6: undefined: "github.com/cs3org/reva/cmd/revad/runtime".RunWithOptions
../ocis-reva/pkg/command/gateway.go:168:7: undefined: "github.com/cs3org/reva/cmd/revad/runtime".WithLogger
../ocis-reva/pkg/command/sharing.go:106:6: undefined: "github.com/cs3org/reva/cmd/revad/runtime".RunWithOptions
../ocis-reva/pkg/command/sharing.go:109:7: undefined: "github.com/cs3org/reva/cmd/revad/runtime".WithLogger
../ocis-reva/pkg/command/sharing.go:109:7: too many errors
make: *** [Makefile:105: bin/ocis] Error 2
```

**Comment by PVince81** [comment]
if I revert "internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/shares.go" the test passes, so there's likely some piece of change that affected the non-public share logic

**Comment by PVince81** [comment]
updated to include https://github.com/cs3org/reva/pull/788

expecting all CI to pass

also since we're updating, we need to add a changelog:
- [ ] add changelog

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/storagepubliclink.go  8
- pkg/flagset/storagepubliclink.go  1
         

Clones added
============
- pkg/command/frontend.go  1
- pkg/command/storagepubliclink.go  9
- pkg/flagset/storagepubliclink.go  4
- pkg/command/gateway.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=5602851&bid=18353755)

**Comment by PVince81** [comment]
added changelog entries

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/storagepubliclink.go  8
- pkg/flagset/storagepubliclink.go  1
         

Clones added
============
- pkg/command/frontend.go  1
- pkg/command/storagepubliclink.go  9
- pkg/flagset/storagepubliclink.go  4
- pkg/command/gateway.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=5602851&bid=18353755)

File: pkg/command/gateway.go [hunk 1]
```diff
@@ -131,16 +134,18 @@ func Gateway(cfg *config.Config) *cli.Command {
...
+											cfg.Reva.StorageOC.MountPath:           cfg.Reva.StorageOC.URL,
+											cfg.Reva.StorageOC.MountID:             cfg.Reva.StorageOC.URL,
+											cfg.Reva.StorageS3.MountPath:           cfg.Reva.StorageS3.URL,
+											cfg.Reva.StorageS3.MountID:             cfg.Reva.StorageS3.URL,
+											cfg.Reva.StorageWND.MountPath:          cfg.Reva.StorageWND.URL,
+											cfg.Reva.StorageWND.MountID:            cfg.Reva.StorageWND.URL,
+											cfg.Reva.StorageCustom.MountPath:       cfg.Reva.StorageCustom.URL,
+											cfg.Reva.StorageCustom.MountID:         cfg.Reva.StorageCustom.URL,
+											"/public/":                             "localhost:10054",
+											"e1a73ede-549b-4226-abdf-40e69ca8230d": "localhost:10054",
```
Comment (PVince81) [inline pkg/command/gateway.go:148]: what is this hash ? should it be a constant ?
what about the port ?

if for later, please raise tickets

**Comment by PVince81** [comment]
@micbar we've updated reva here so I added the usual changelog, in case you want to review just that bit

**Pull request merged**